torch/_dynamo/variables/tensor.py

```diff
@@ -644,6 +645,57 @@ def redistribute_fn_with_prim_types(x):
                 ),
                 **options,
             )
+        elif name == "register_hook":
+            # see [On tensor.register_hook]
+            assert len(args) == 1
+            fn_var = args[0]
+
+            if isinstance(fn_var, variables.NestedUserFunctionVariable):
+                # NestedUserFunctionVariable don't carry their fn, but reconstruction builds it
+                # This should not be onerous to support when needed.
+                unimplemented("NYI - lambda variables as hooks")
+            elif isinstance(fn_var.fn, functools.partial):
+                fn = fn_var.fn
+                name = fn_var.fn.func.__name__
+            else:
+                fn = fn_var.fn
+                name = fn_var.fn.__name__
+
+            handle = self.as_proxy().node.meta["example_value"].register_hook(fn)
```

Comment:
You are unpredictably calling register_hook on either a fake or real tensor. If it's a fake tensor, this is doing nothing, because we are never going to backwards trough it. Is that intended?

Reply:
We just need a way to get the handle out, to recover the name. Super open to feedback on how to do this better tho.

Reply:
But you don't need a handle. You know that `register_hook` is always going to create a fresh handle and so you just make a fresh handle VT. You don't need an actual handle on the inside, and it's better not to have one because it isn't going to do anything useful...

Reply:
good point